include/gsl/span

```diff
@@ -627,71 +677,87 @@ private:
 
     // The rest is needed to remove unnecessary null check
     // in subspans and constructors from arrays
-    constexpr span(KnownNotNull ptr, index_type count) : storage_(ptr, count) {}
+    constexpr span(KnownNotNull ptr, size_type count) : storage_(ptr, count) {}
```

Comment:
is this needed?

Reply:
we should probably keep this for perf gains from not checking the nullness of the pointers.